backend/game/engine/player_queue.py

```diff
@@ -0,0 +1,45 @@
+import itertools
+
+
+class GamePlayers():
+
+    def __init__(self):
+        self.all_players = []
+        self.dead_players = []
```

Comment:
You don't need to keep lists of dead and alive players. Instead, you can iterate through and pass the players that are dead. It does not make sense to keep multiple lists as they can get out of sync very quickly. That's a lot of headache for no real upsides.